examples/catalog/lib/custom_a11y_traversal.dart

```diff
@@ -0,0 +1,342 @@
+  const RowColTraversal({this.rowOrder, this.columnOrder, this.child});
+
+  final int rowOrder;
+  final int columnOrder;
+  final Widget child;
+
+  /// Builds a widget hierarchy that wraps [child].
+  ///
+  /// This function expresses the sort keys as a hierarchy, but it could just as
+  /// easily have been a flat list:
+  ///
+  /// ```
+  ///  Widget build(BuildContext context) {
+  ///    return new Semantics(
+  ///      sortOrder: new SemanticsSortOrder(
+  ///        keys: <SemanticsSortKey>[
+  ///          new OrdinalSortKey(columnOrder.toDouble()),
+  ///          new OrdinalSortKey(rowOrder.toDouble())
+  ///        ],
+  ///      ),
+  ///      child: child,
+  ///    );
+  ///  }
+  /// ```
+  ///
+  /// The resulting order is the same.
+  @override
+  Widget build(BuildContext context) {
+    return new Semantics(
+      sortKey: new OrdinalSortKey(columnOrder.toDouble()),
+      child: new Semantics(
+        sortKey: new OrdinalSortKey(rowOrder.toDouble()),
+        child: child,
+      ),
+    );
+  }
+}
+
+// --------------- Component widgets ---------------------
+
+/// A Button class that wraps an [IconButton] with a [RowColTraversal] to
+/// set its traversal order.
+class SpinnerButton extends StatelessWidget {
+  const SpinnerButton({
+    Key key,
+    this.onPressed,
+    this.icon,
+    this.rowOrder,
+    this.columnOrder,
+    this.field,
+    this.increment
+  }) : super(key: key);
+
+  final VoidCallback onPressed;
+  final IconData icon;
+  final int rowOrder;
+  final int columnOrder;
+  final Field field;
+  final bool increment;
+
+  @override
+  Widget build(BuildContext context) {
+    final String label = '${increment ? 'Increment' : 'Decrement'} ${_fieldToName(field)}';
+
+    return new RowColTraversal(
+      rowOrder: rowOrder,
+      columnOrder: columnOrder,
+      child: new Center(
+        child: new IconButton(
+          icon: new Icon(icon),
+          onPressed: onPressed,
+          tooltip: label,
+        ),
+      ),
+    );
+  }
+}
+
+/// A text entry field that wraps a [TextField] with a [RowColTraversal] to
+/// set its traversal order.
+class FieldWidget extends StatelessWidget {
+  const FieldWidget({
+    Key key,
+    this.rowOrder,
+    this.columnOrder,
+    this.onIncrease,
+    this.onDecrease,
+    this.value,
+    this.field,
+  }) : super(key: key);
+
+  final int rowOrder;
+  final int columnOrder;
+  final VoidCallback onDecrease;
+  final VoidCallback onIncrease;
+  final int value;
+  final Field field;
+
+  @override
+  Widget build(BuildContext context) {
+    final String stringValue = '${_fieldToName(field)} $value';
+    final String increasedValue = '${_fieldToName(field)} ${value + 1}';
+    final String decreasedValue = '${_fieldToName(field)} ${value - 1}';
+
+    return new RowColTraversal(
+      rowOrder: rowOrder,
+      columnOrder: columnOrder,
+      child: new Center(
+        child: new Semantics(
+          onDecrease: onDecrease,
```

Comment:
Oh nice! So in a11y mode you have two ways to manipulate the number: either use decrease/increase actions or click the up/down button!